app/controllers/api/v1/visualization/clusters_controller.rb

```diff
@@ -15,6 +16,8 @@ class ClustersController < ApiBaseController
         before_action :set_current_api_user!
         before_action :set_study
         before_action :check_study_view_permission
+        before_action :check_api_cache!
+        after_action :write_api_cache!
```

Comment:
it's a bummer we have to explicitly specify these to get the ordering we want.  Can you get the same effect by just doing the include down here?

Reply:
As we move forward with extracting logic out into the API, I imagine both of these controllers getting much larger.  With that in mind, I decided I wanted these methods declared explicitly so that we could control which endpoints get cached using `:only` or `:except`.  There quite possibly could be methods that we don't want cached in here.